### PR TITLE
Update to Bevy 0.14

### PR DESCRIPTION
**Pull request by BD103**
This updates the `main` branch to use Bevy 0.14, though it does not touch the other branches. As part of this change, I also added `publish = false` to `Cargo.toml` to prevent it from accidentally being published on <https://crates.io>.

### CODE REVIEW

**Review by janhohenheim — APPROVED** [review]

**Comment by BD103** [comment]
Though I technically have merge rights, I'm going to wait for a maintainer or @JMS55 to approve this. :)

**Pull request merged**
